contracts/validator-manager/ValidatorMessages.sol

```diff
@@ -146,48 +146,21 @@ library ValidatorMessages {
         pure
         returns (bytes memory)
     {
-        // The formula for the length in the comment above is VM agnostic, but in EVM
-        // the address length is always 20 bytes so the constant term is 72 + 20 = 92.
-        bytes memory res = new bytes(92 + subnetConversionData.initialValidators.length * 88);
-        // Pack the convertSubnetTx ID
-        for (uint256 i; i < 32; ++i) {
-            res[i] = subnetConversionData.convertSubnetTxID[i];
-        }
-        // Pack the validatorManagerBlockchainID
-        for (uint256 i; i < 32; ++i) {
-            res[i + 32] = subnetConversionData.validatorManagerBlockchainID[i];
-        }
-        // Pack the ADDRESS_LENGTH
-        for (uint256 i; i < 4; ++i) {
-            res[i + 64] = bytes1(uint8(20 >> (8 * (3 - i))));
-        }
-        // Pack the address
-        bytes20 addrBytes = bytes20(subnetConversionData.validatorManagerAddress);
-        for (uint256 i = 0; i < 20; ++i) {
-            res[i + 68] = addrBytes[i];
-        }
-
-        // Pack the initial validators length
-        uint32 ivLength = uint32(subnetConversionData.initialValidators.length);
-        for (uint256 i; i < 4; ++i) {
-            res[i + 88] = bytes1(uint8(ivLength >> (8 * (3 - i))));
-        }
-
+        // Hardcoded 20 is for length of the managerAddress
```

Comment:
can you add a comment about why encode packed is used instead of preallocated array

Reply:
Hmm I can but we already use encodePacked anyway and results of my experiment show it to just be empirically true not absolutely true.

Reply:
yup that's fine, just a note that we at least took a look at the other approach previously, so we know when we look back